### PR TITLE
Blog app - integration specs for views and fix n+1 problems

### PR DESCRIPTION
**Pull request by Ouail-01**
> ### In this pull request I did the following change: 

- [X] Write integration tests with Capybara gem.
- [X] Secure app from n+1 problems.
- [X] Used Capybara to write integration tests for:
- User index page
- User show page
- Post index page
- Post show page

### CODE REVIEW

File: spec/system/user_index_spec.rb [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+      visit '/users'
+      @users.each do |user|
+        expect(page).to have_content "Number of posts: #{user.PostsCounter}"
+      end
+    end
+
+    it 'shows the username of all other users' do
+      visit '/users'
+      expect(page).to have_content(@user1.Name)
+      sleep(2)
```
Comment (thecodechaser) [inline spec/system/user_index_spec.rb:43]: - [x]  Please make sure to write the below tests for the user index page as requested in the project requirements.
  - `Tests that redirect to the user's show page when you click on a user.`

File: spec/system/user_show_spec.rb [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    expect(page).to have_content("Number of posts: #{@user1.PostsCounter}")
+    # sleep(2)
+  end
+
+  it 'shows theuser\'s bio ' do
+    visit "/users/#{@user1.id}"
+    expect(page).to have_content(@user1.Bio)
+    # sleep(2)
+  end
+
```
Comment (thecodechaser) [inline spec/system/user_show_spec.rb]: - [x]  Please make sure to write the below tests for the user show page as requested in the project requirements.
  - `Tests that asserts you can see a button that lets me view all of a user's posts`
  - `Tests that redirect to the user's post's index page when you click on a see all posts.`

File: spec/feature/post_index_spec.rb [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  end
+
+  scenario 'display the post body' do
+    expect(page).to have_content(@first_post.Text)
+  end
+
+  scenario 'display the first comment on a post' do
+    expect(page).to have_content('Microverse')
+  end
+
```
Comment (thecodechaser) [inline spec/feature/post_index_spec.rb:34]: - [x] Please make sure to write the below tests for the post index page as requested in the project requirements.
  - `Tests that asserts you can see the user's profile picture.`
  - `Tests that asserts you can see the user's username.`
  - `Tests that asserts you can see the number of posts the user has written.`

File: spec/feature/post_ruby_spec.rb [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+  scenario 'displays the post title' do
+    expect(page).to have_content(@first_post.Title)
+  end
+
+  scenario 'shows the person who wrote the post' do
+    expect(page).to have_content('Ouail')
+  end
+
+  it 'can see the username of each commentor.' do
```
Comment (thecodechaser) [inline spec/feature/post_ruby_spec.rb]: - [x] Please change this file name to `post_show_spec.rb`, It will help to understand what tests it contains.

**Review by thecodechaser — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ 

## Hi @Ouail-01, and @Adholah96👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
- No linter errors. :heavy_check_mark:
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️



## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by zieeco — APPROVED** [review]
# status: APPROVED :green_circle: :100: 

Hi, @Team, 




Good job so far 👍 Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


## Highlights ⚡ 

✅ Descriptive PR 👍 
✅ Nice implementing the changes 🔥 
✅ Linters are passing 👏 


## Optional suggestions

- N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._



Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**